types/justifiedgallery/index.d.ts

```diff
@@ -0,0 +1,54 @@
+// Type definitions for justifiedGallery 3.8
+// Project: https://github.com/miromannino/Justified-Gallery
+// Definitions by: Matt Smith <https://github.com/mattsmithcode>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="jquery"/>
+
+declare namespace JustifiedGallery {
+    function justifiedGallery(root: Window, jQuery?: JQueryStatic): JQueryStatic | JQuery | HTMLElement;
```

Comment:
so, I don't think the function itself should be declared here, maybe just declare the function type and use it when augmenting the JQuery interface.
Otherwise there will be a namespace `JustifiedGallery` available with a function, `JustifiedGallery.justifiedGallery`, and I don't think that accurately reflects the way the underlying JS works, but I might be wrong.

Reply:
@gabritto not entirely following what you mean by declaring the function type. I do agree with the function not being declared in the namespace although I'm not sure how best to define the module. (First time contributing to DT so suggestions are certainly appreciated!)

Reply:
@mattsmithcode try something along:
```ts
declare namespace JustifiedGallery {
    interface Settings {
       ...
    }

    interface CaptionSettings {
      ...
    }
}

declare function justifiedGallery(root: Window, jQuery?: JQueryStatic): JQueryStatic | JQuery | HTMLElement;

declare global {
    interface JQuery<TElement = HTMLElement> {
        justifiedGallery(settings?: JustifiedGallery.Settings): JQuery;
    }
}

export = justifiedGallery;
```

Reply:
Yep, that's got it!
Thanks